sycl/include/CL/sycl/exception.hpp

```diff
@@ -36,15 +36,14 @@ struct exception {
   shared_ptr_class<context> MContext;
 
 protected:
-  exception(const char *Msg, int CLErr = CL_SUCCESS,
+  exception(const char *Msg, const cl_int CLErr = CL_SUCCESS,
             shared_ptr_class<context> Context = nullptr)
-      : MMsg(string_class(Msg) + " " +
-             ((CLErr == CL_SUCCESS) ? "" : OCL_CODE_TO_STR(CLErr))),
-        MCLErr(CLErr), MContext(Context) {}
+      : exception(string_class(Msg), CLErr, Context) {}
 
-  exception(const string_class &Msg, int CLErr = CL_SUCCESS,
+  exception(const string_class &Msg, const cl_int CLErr = CL_SUCCESS,
             shared_ptr_class<context> Context = nullptr)
-      : exception(Msg.c_str(), CLErr, Context) {}
+      : MMsg(Msg + " " + OCL_CODE_TO_STR(CLErr)), MCLErr(CLErr),
```

Comment:
If `CLErr == CL_SUCCESS`, the message is going to look weird.

Reply:
Hm, do you agree with it?
`MMsg(Msg + ((CLErr == CL_SUCCESS) ? "" :  " " + OCL_CODE_TO_STR(CLErr)))`

Reply:
That should work. You probably need to wrap `" " + OCL_CODE_TO_STR(CLErr)` into an `std::string`, because you cannot concatenate C strings.